refact executeCallback to avoid this

BananaDeveloper opened the pull request:
Refactored executeCallback function to avoid using this.

Updated functions that use executeCallback.

Added a sample test file. You can expand it.

BananaDeveloper committed:
refact executeCallback to avoid using this

BananaDeveloper committed:
changed test globbing path

BananaDeveloper committed:
removed pass with no test flag from jest npm script

BananaDeveloper committed:
unit test file for executeCallback helper

SimplySayHi reviewed (changes requested):
function(s) passed to executeCallback must run with the form instance as context.
It can be useful to set the context for all the functions ( onValidation, beforeSubmit, etc ) directly inside the constructor so that we don't have to add it in multiple files.

Commit:
removed useless self variable

Commit:
binding this to every field option and form option passed to execute callback

BananaDeveloper commented:
> function(s) passed to executeCallback must run with the form instance as context.
> It can be useful to set the context for all the functions ( onValidation, beforeSubmit, etc ) directly inside the constructor so that we don't have to add it in multiple files.

Done! 
I am bidning context ony for the options that are passed to executCallback. 
If any other options must be bound please send me a complete list.

Commit:
Merge branch 'refactoring-functions-to-avoid-this' into refact-execute-callback

SimplySayHi committed:
Update constructor.js

callbacks in fieldOptions/formOptions are arrays so we need to map each function inside of them

BananaDeveloper commented:
@SimplySayHi thanks for the fixes :smile:
